# DO NOT EDIT 'install.R'; instead, edit 'install.R.in' and
# use 'rake' to generate 'install.R'.

pkgs <- c('affy',
 'affyio',
 'airway',
 'ALL',
 'annotate',
 'AnnotationDbi',
 'AnnotationFilter',
 'AnnotationHub',
 'apeglm',
 'beachmat',
 'Biobase',
 'BiocGenerics',
 'BiocParallel',
 'BiocStyle',
 'BiocVersion',
 'biomaRt',
 'Biostrings',
 'BSgenome',
 'BSgenome.Hsapiens.UCSC.hg19',
 'BSgenome.Hsapiens.UCSC.hg19.masked',
 'caTools',
 'clusterExperiment',
 'clusterProfiler',
 'curatedMetagenomicData',
 'curatedTCGAData',
 'DECIPHER',
 'DelayedArray',
 'DelayedMatrixStats',
 'DESeq',
 'DESeq2',
 'DOSE',
 'downloader',
 'edgeR',
 'edgeR',
 'EnrichmentBrowser',
 'EnsDb.Hsapiens.v79',
 'EnsDb.Hsapiens.v86',
 'EnsDb.Mmusculus.v79',
 'ensembldb',
 'fgsea',
 'gdata',
 'genefilter',
 'geneplotter',
 'GenomeInfoDb',
 'GenomicAlignments',
 'GenomicDataCommons',
 'GenomicFeatures',
 'GenomicRanges',
 'GEOquery',
 'Glimma',
 'GOSemSim',
 'gProfileR',
 'graph',
 'graph',
 'GSEABase',
 'Gviz',
 'HDF5Array',
 'hgu133a.db',
 'hgu95av2.db',
 'HMP16SData',
 'Homo.sapiens',
 'httr',
 'hugene20sttranscriptcluster.db',
 'impute',
 'IRanges',
 'knitr',
 'limma',
 'MultiAssayExperiment',
 'multtest',
 'Mus.musculus',
 'org.Hs.eg.db',
 'OrganismDbi',
 'PharmacoGx',
 'pheatmap',
 'phyloseq',
 'piano',
 'plyranges',
 'preprocessCore',
 'ProtGenerics',
 'qvalue',
 'RaggedExperiment',
 'RBGL',
 'RCurl',
 'RCy3',
 'recount',
 'regioneR',
 'ReportingTools',
 'Rgraphviz',
 'rhdf5',
 'Rhdf5lib',
 'rmarkdown',
 'RNAseq123',
 'Rsamtools',
 'rtracklayer',
 'S4Vectors',
 'SingleCellExperiment',
 'splatter',
 'stringr',
 'SummarizedExperiment',
 'survival',
 'survminer',
 'sva',
 'TCGAbiolinks',
 'TCGAutils',
 'TxDb.Hsapiens.UCSC.hg19.knownGene',
 'TxDb.Hsapiens.UCSC.hg38.knownGene',
 'tximport',
 'UpSetR',
 'VariantAnnotation',
 'VennDiagram',
 'XVector',
 'zinbwave',
 'zlibbioc',
 'systemPipeR',
 'systemPipeRdata'
)

ap.db <- available.packages(contrib.url(BiocManager::repositories()))
ap <- rownames(ap.db)
fnd <- pkgs %in% ap
pkgs_to_install <- pkgs[fnd]

ok <- BiocManager::install(pkgs_to_install, update=FALSE, ask=FALSE) %in% rownames(installed.packages())

if (!all(fnd))
    message("Packages not found in a valid repository (skipped):\n  ",
            paste(pkgs[!fnd], collapse="  \n  "))

suppressWarnings(BiocManager::install(update=TRUE, ask=FALSE))
